src/Place/AbstractPlace.php

```diff
@@ -72,10 +72,36 @@ public function getThreshold()
      * @var array the failures, timeout and treshold
      *
      * @return self
+     *
+     * @throws InvalidPlaceException
      */
     public static function fromArray(array $settings)
     {
-        return new static($settings[0], $settings[1], $settings[2]);
+        static::validateSettings($settings);
```

Comment:
I don't like static methods :/
And check is already done in validate method.

```php
        return new static(
            isset($settings['failures']) ? $settings['failures'] : null,
            isset($settings['timeout']) ? $settings['timeout'] : null,
            isset($settings['threshold']) ? $settings['threshold'] : null
        );
```

Reply:
Alright, I kept the static `fromArray` method but moved the validation inside it
I kept this method for backward compatibility.. but since we allowed BC breaks maybe I should just remove it (it is not used in the factory any more)

Reply:
The validation is useless since it's already done here:
https://github.com/PrestaShop/circuit-breaker/blob/ee56233751dd7e99348c02042cf403f813f4d773/src/Place/AbstractPlace.php#L33

And already generate an invalidSettings exceptions.
https://github.com/PrestaShop/circuit-breaker/blob/ee56233751dd7e99348c02042cf403f813f4d773/src/Place/AbstractPlace.php#L113

Reply:
This is not the same validation, `fromArray` validates that the array is correctly formatted (has the right fields), whereas `validate` validates that the value used are correct (integer, positive, ...)
But again maybe we can simply drop this `fromArray` method

Reply:
I think we can keep the `fromArray` method and only check if data are correctly validated